### PR TITLE
feat(connector-besu, connector-quorum): filesystem replaced by keychain

### LINKED ISSUE #678: feat(connector-besu): replace filesystem for keychain
**Is your feature request related to a problem? Please describe.**
In connector Besu and connector Quorum, the invokeContractV2 use the filesystem to read the JSON of the smartcontracts.

**Describe the solution you'd like**
Replace de filesystem for keychain or consortium database.

**Acceptance criteria**
I should be able to read the JSON and rewrite it if necessary with the new networkId and the contract address.

### LINKED ISSUE #680: fix(connector-quorum): check networks field exists in the contract compilation file
**Is your feature request related to a problem? Please describe.**
Actually, if the networks field does not exist in the contract compilation file, the functionality added in issue #612 will fail.

**Describe the solution you'd like**
In the deploy function, add a check to verify if the contract compilation file has the networks field. If not, return an error.
This is also necessary for the besu connector.

### PR DESCRIPTION
**Pull request by jordigiam**
Resolves #678 
Resolves #680

### CODE REVIEW

**Review by petermetz — CHANGES_REQUESTED** [review]
@jordigiam When we generate the changelogs at release time, with the current commit message, this will show up as besu getting the keychain support, but there will be nothing there about quorum. Is there any chance that you could either break it up into two commits (it's OK to keep the PR as is, with 2 commits because they are doing the same change anyway) like
Commit 1 `feat(connector-besu): filesystem replaced by keychain`
Commit 2 `feat(connector-quorum): filesystem replaced by keychain`

Or you could also leave it in one commit and just edit the scope to include both besu and quorum like
`feat(connector-besu,connector-quorum): filesystem replaced by keychain`
Assuming it's not over the character limit this would work as well. Also any other solution that makes sure that the generated changelogs will mention both quorum and besu would work for me, but I only know of the two options above to make it so.

File: packages/cactus-plugin-ledger-connector-besu/src/main/typescript/plugin-ledger-connector-besu.ts [hunk 1]
```diff
@@ -255,53 +252,92 @@ export class PluginLedgerConnectorBesu
...
+          req.keychainId,
+        );
+        Checks.truthy(
+          keychainPlugin,
+          `${fnTag} keychain for ID:"${req.keychainId}"`,
         );
-        if (contractJson.contractName != contractName) {
+        if (!keychainPlugin.has(contractName)) {
           throw new Error(
             `${fnTag} Cannot create an instance of the contract because the contractName and the contractName of the JSON doesn't match`,
```
[the remark this answers is not in the document]
Comment (jordigiam) [reply packages/cactus-plugin-ledger-connector-besu/src/main/typescript/plugin-ledger-connector-besu.ts:308]: Done!

**Review by petermetz — CHANGES_REQUESTED** [review]
@jordigiam Just tracking/noting down for the record here what we discussed on the pair programming session today => please make sure that the request can always override whatever is in `this.contracts` by using the keychain instead.

**Review by petermetz — APPROVED** [review]
LGTM

Note to self: We need to put in some time to design something that's stateless so that the contracts map can be eliminated entirely (somehow, but I don't know any better right now either except for the consortium database, but that's also not a mature design for contract storage so back to my initial point about putting time into some design...)

One thing I noticed is that we could expose a get network ID endpoint on the web3 flavored connectors (besu, quorum) so that we can build on that in the future by requiring the caller to fill the network ID themselves and freeing the connector from having to do that altogether.

**Review by takeutak — APPROVED** [review]

**Pull request merged**